core/modules/module_map_data.rs

```diff
@@ -516,11 +515,11 @@ impl ModuleMapData {
   #[cfg(test)]
   pub fn assert_module_map(&self, modules: &Vec<ModuleInfo>) {
     let data = self;
-    // There's always one internal `deno_core` ES module loaded, so +1 here.
-    assert_eq!(data.handles.len(), modules.len() + 1);
-    assert_eq!(data.info.len(), modules.len() + 1);
-    assert_eq!(data.next_load_id as usize, modules.len());
-    assert_eq!(data.by_name.len(), modules.len() + 1);
+    // There's always two internal `deno_core` ES module loaded, so +2 here.
+    assert_eq!(data.handles.len(), modules.len() + 2);
+    assert_eq!(data.info.len(), modules.len() + 2);
+    assert_eq!(data.next_load_id as usize, modules.len() + 1);
```

Comment:
```suggestion
    assert_eq!(data.next_load_id as usize, modules.len());
```

Reply:
I'm not sure about that. I think we're leaking some detail here. I want to investigate this `next_load_id` before I merge this PR.

Reply:
IIUC `next_load_id` is only tracked for non internal modules.